functionary/ui/templates/core/team_detail.html

```diff
@@ -1,60 +1,48 @@
-        </nav>
-    </div>
-    <div class="column">
-        <div class="block">
-            <h1 class="title is-1">
-                <p>
-                    <i class="fa fa-users"></i>
-                    <span>{{ team.name }}</span>
-                </p>
-            </h1>
-        </div>
-        <div class="column has-addons">
-            <div class="field">
-                <label class="label" for="envs">
-                    <i class="fa fa-building"></i>&nbsp;Environments:
-                </label>
-                <ul id="envs">
+    <nav aria-label="breadcrumb">
+        <ol class="breadcrumb">
+            <li class="breadcrumb-item">
+                <a class="text-decoration-none" href="{% url 'ui:team-list' %}">Team List</a>
+            </li>
+            <li class="breadcrumb-item active" aria-current="page">{{ team.name }}</li>
+        </ol>
+    </nav>
+    <h1 class="mt-2 mb-4">
+        <i class="fa fa-users"></i>
+        <span>{{ team.name }}</span>
+    </h1>
+    <div class="vstack gap-2 my-2">
+        <div>
+            <strong class="fs-4"><i class="fa fa-building fa-sm fa-fw me-1"></i>Environments</strong>
+            <div class="ms-4 my-2">
+                <div id="envs" class="list-group list-group-flush">
                     {% for env in environments %}
-                        <li class="block ml-4">
-                            <a href="{% url 'ui:environment-detail' env.id %}">{{ env.name }}</a>
-                            <br/>
-                            {{ env.description }}
-                        </li>
+                        <a href="{% url 'ui:environment-detail' env.id %}"
+                           class="list-group-item list-group-item-action">
+                            <p class="mb-0">{{ env.name }}</p>
+                        </a>
                     {% endfor %}
-                </ul>
+                </div>
             </div>
-            <div class="field">
-                {% if not team_create_perm %}
-                    <label class="label" for="users">
-                        <i class="fa fa-user"></i>&nbsp;Users:
-                    </label>
-                    <ul id="users">
+        </div>
+        {% if not request.session.user_can_update_team %}
+            <div>
+                <strong class="fs-4"><i class="fa fa-user fa-sm fa-fw me-1"></i>Users</strong>
+                <div class="ms-4 my-2">
+                    <div id="users" class="list-group list-group-flush">
                         {% for u in user_details %}
-                            <li class="block ml-4">
-                                <p>
-                                    {{ u.first_name }} {{ u.last_name }} <span class="has-text-grey-light">({{ u.username }})</span>
-                                </p>
-                            </li>
+                            <p class="list-group-item fw-bold mb-0">
+                                {{ u.first_name }} {{ u.last_name }}
```

Comment:
This was wrong before and not a result of these changes, but might as well fix it for now.  The `u` here is not the user object, it's some other object that contains role info in addition to the user.  So each `u.` would have to be `u.user.`.

I wonder if the better thing to do here would be remove the permissions check here and then just wrap the various buttons in the `team_management.html` with the permissions checks.  That way we don't have separate rendering logic for this info, it's just that the editing is disabled if you don't have access.